redfish-core/lib/assembly.hpp

```diff
@@ -435,6 +435,96 @@ inline void
         "xyz.openbmc_project.Association", "endpoints");
 }
 
+void startOrStopADCSensor(const bool start,
+                          const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    std::string method{"StartUnit"};
+    if (!start)
+    {
+        method = "StopUnit";
+    }
+
+    crow::connections::systemBus->async_method_call(
+        [asyncResp](const boost::system::error_code ec) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR << "Failed to start or stop ADCSensor:" << ec;
+                messages::internalError(asyncResp->res);
+                return;
+            }
+            messages::success(asyncResp->res);
+        },
+        "org.freedesktop.systemd1", "/org/freedesktop/systemd1",
+        "org.freedesktop.systemd1.Manager", method,
+        "xyz.openbmc_project.adcsensor.service", "replace");
+}
+
+void doBatteryCM(const std::string& assembly, const bool readyToRemove,
+                 const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    if (readyToRemove)
+    {
+        // Stop the adcsensor service so it doesn't monitor the battery
+        startOrStopADCSensor(false, asyncResp);
+        return;
+    }
+
+    // Find the service that has the OperationalStatus iface, set the
```

Comment:
This comment is helpful.

When readyToRemove is true, we are stopping the adcsensor, that code is above on line 468. Makes sense. 

When readyToRemove is false, this is the only time we get in here. 
readyToRemove =  false is done after the CM.. 
So when readyToRemove is false, we set functional to true and start the ADC sensor. Makes sense.